Std/Data/HashMap/Lemmas.lean

```diff
@@ -5,11 +5,21 @@ Authors: Scott Morrison
 -/
 import Std.Data.HashMap.Basic
 import Std.Util.ProofWanted
+import Std.Data.Array.Lemmas
 
 namespace Std.HashMap
 
-@[simp] proof_wanted empty_find? [BEq α] [Hashable α] {a : α} :
-    (∅ : HashMap α β).find? a = none
+theorem empty_def [BEq α] [Hashable α] : (∅ : HashMap α β) = HashMap.empty :=
+  rfl
+
+@[simp] theorem mkArray_getElem {α : Type u} {v: α} {n i : Nat}
+    (h : i < Array.size (mkArray n v)) :
+    (mkArray n v)[i]'h = v :=
+  List.eq_of_mem_replicate (Array.getElem_mem_data _ h)
```

Comment:
This should be moved to `Data/Array/Lemmas` and renamed `getElem_mkArray`. See #544:
https://github.com/leanprover/std4/blob/eb1f7b8dc292bd412ecc80c0ba8219922cc67d7a/Std/Data/Array/Lemmas.lean#L33-L34